lib/client/api.go

```diff
@@ -1088,6 +1088,32 @@ func Status(profileDir, proxyHost string) (*ProfileStatus, []*ProfileStatus, err
 	return profileStatus, others, nil
 }
 
+// FetchRoleSet fetches a user's roles for a specified cluster.
+func FetchRoleSet(ctx context.Context, log *logrus.Entry, cluster auth.ClientI, profile *ProfileStatus) (services.RoleSet, error) {
```

Comment:
A few asks:

1. Should this be under `lib.services` instead, alongside `services.FetchRoles`?
2. Could we use an interface smaller than `auth.ClientI`? An interface declared specifically for this method is perfectly fine.
3. Could you add a small test for this func?

Reply:
> 1. Should this be under `lib.services` instead, alongside `services.FetchRoles`?

Good point, thanks! I wasn't sure where to put it so I used my default heuristic of "stuff that talks with the cluster that needs to be used by both tsh and Connect should go to `lib/client`.

> 2\. Could we use an interface smaller than `auth.ClientI`? An interface declared specifically for this method is perfectly fine.

Sure.

> 3\. Could you add a small test for this func?

Yes, I'll try. 🥲 Though I'm already pinned down by writing tests under another PR and it slows me down considerably, having little experience with writing tests in a statically typed lang. But I had to get over this hurdle at some point anyway, it's just that the timing of a week before the release is unfortunate.

Reply:
Give me a nudge if you need some help. Alternatively, since it's just a move, I'm happy to take the tests in a follow up PR if you are too swamped now (but let's try to have that happen sometime).